training/train.py

```diff
@@ -302,30 +302,48 @@ def create_git_pr(s3_client, model_version, recall_at_30):  # pragma: no cover
     prev_hyperparams = s3_client.read_json_file(k)
 
     # Convert the json description to string
-    description = json.dumps(prev_hyperparams).replace('"', '\\"')
+    description = 'Previous hyper parameters :: ' + json.dumps(prev_hyperparams).replace('"', '\\"')
 
     prev_recall = prev_hyperparams.get('recall_at_30', 0.55)
+    _logger.info('create_git_pr:: Prev => Model {}, Recall {}  Curr => Model {}, Recall {}'.format(
+        previous_version, prev_recall, model_version, recall_at_30))
     if recall_at_30 >= prev_recall:
         try:
             # Invoke bash script to create a saas-analytics PR
-            t = subprocess.Popen(['sh', 'rudra/utils/github_helper.sh', 'f8a-hpf-insights.yaml',
-                                  'MODEL_VERSION', str(model_version), description],
-                                 shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            # Wait for the subprocess to get over
-            t.wait(60)
-            if t.returncode == 0:
-                _logger.info("Successfully created a PR")
+            t1 = subprocess.Popen(['wget', '-v',
```

Comment:
Can you refactor this to a method named call_sub_process and then pass the parameters required?
```suggestion
            t1 = call_sub_process(['wget', '-v', ....])
            t2 = call_sub_process([chmod, +x ....])
            t3 = call_sub_process(['./github_helper.sh', ....]
```
That method can do the error handling for each call. This will help clean up a lot of repetition.

Reply:
I thought of enabling the flow first, as this all flow will change drastically when I pickup code refactoring task following the current task. However, suggestion looks and straight forward, let me fix it.